File: src/arp/arp.cpp
#include "arp.h"

CppMacAddress cpp_send_arp(uint32_t remote_ip) {
  CppMacAddress mac_addr = {0};

  ULONG mac_buffer[2] = {0};
  ULONG mac_size = sizeof(mac_buffer);
  DWORD err_code = SendARP(remote_ip, 0, mac_buffer, &mac_size);

  if (NO_ERROR != err_code) {
    return mac_addr;
  }

  if (6 != mac_size) {
    return mac_addr;
  }

  memcpy(mac_addr.addr, mac_buffer, 6);
  return mac_addr;
}

File: src/samba/arp/arp.rs
extern crate libc;

#[link(name="arp", kind = "static")]
extern {
    #[repr(C)]
    struct CppMacAddress {
        addr: [libc::uint_8; 6],
    }

    fn cpp_send_arp(remote_ip: uint32_t) -> CppArpInfo;
}

fn send_arp(remote_ip: u32) -> [u8; 6] {
    let mac_addr = cpp_send_arp(remote_ip);
    return mac_addr.addr;
}

fn send_arp(remote_ip: &str) -> [u8; 6]  {
    return send_arp();
}
